src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -181,6 +182,17 @@ suggestAction dflags packageExports ideOptions parsedModule text diag = concat
     ] ++
     suggestFillHole diag                   -- Lowest priority
 
+suggestDisableWarning :: Diagnostic -> [(T.Text, [TextEdit])]
+suggestDisableWarning Diagnostic{..}
+    | Just (StringValue (showFlag -> Just w)) <- _code =
+        pure
+            ( "Disable \"" <> w <> "\" warnings"
+            , [TextEdit (Range (Position 0 0) (Position 0 0)) $ "{-# OPTIONS_GHC -Wno-" <> w <> " #-}\n"]
```

Comment:
Alternatively, we could try to insert beneath any existing pragmas. If e.g. we have an `{-# OPTIONS_GHC -Wall #-}` in the file, then inserting this pragma at the top may have no effect.

Reply:
Does the order of ghc options matter? I'm not quite sure
Either way, I would prefer if this pragma was inserted beneath the existing pragmas, specially because it's quite common to have a copyright notice *above* the pragmas

Reply:
Yeah, I've just checked. With:
```hs
{-# OPTIONS_GHC -Wno-unused-imports #-}
{-# OPTIONS_GHC -Wall #-}
```

the first pragma has no effect.

Reply:
Is there a clean way to find the pragma locations? AFAICT, they don't appear in `ParsedModule`.

Best thing I can think of otherwise is to get the location of the module declaration (falling back to first import, then first declaration, if there isn't one), and insert immediately before that. Or even backtrack from there by looking at the raw text for lines beginning with `{-#`...

Reply:
Presumably we want pretty much the same logic for adding language pragmas? Currently those also go right to the top of the file. So perhaps this belongs in a follow-up PR.

Reply:
From the parsed module you can get the line for the `module Foo where...` declaration, and just put the new pragmas above it.